### PR TITLE
EdgeQL constant extraction

### PR DESCRIPTION
**Pull request by tailhook**
This only extracts integers. Other types should be easier as most special cases are on integers.

### CODE REVIEW

File: edb/server/mng_port/edgecon.pyx [hunk 1]
```diff
@@ -736,12 +754,19 @@ cdef class EdgeConnection:
         object io_format,
         bint expect_one,
         uint64_t implicit_limit,
-    ):
+    ) -> CompiledQuery:
         if self.debug:
             self.debug_print('PARSE', eql)
 
+        entry = rewrite(eql)
```
Comment (elprans) [inline edb/server/mng_port/edgecon.pyx:761]: I think "preparse" (tokenize+extract args) would be a better term for this stage.  Also, `Entry` is nondescript. `PreparsedQuery` seems better.

File: edb/server/mng_port/edgecon.pyx [hunk 2]
```diff
@@ -785,9 +811,14 @@ cdef class EdgeConnection:
 
         if not cached and query_unit.cacheable:
             self.dbview.cache_compiled_query(
-                eql, io_format, expect_one, implicit_limit, query_unit)
+                entry.key(), io_format, expect_one, implicit_limit, query_unit)
 
-        return query_unit
+        return CompiledQuery(
```
[the remark this answers is not in the document]
Comment (tailhook) [reply edb/server/mng_port/edgecon.pyx:816]: `QueryUnit` comes from compiler. Things we add here come from server itself. It's ugly to modify structure with stubs got from compiler.

File: edb/server/mng_port/edgecon.pyx (hunk 1 above)
Comment (tailhook) [reply edb/server/mng_port/edgecon.pyx:761]: Well we have [preparser module](https://github.com/edgedb/edgedb/blob/master/edb/edgeql-parser/src/preparser.rs) which is basically statement delimiter.

Comment (elprans) [reply edb/server/mng_port/edgecon.pyx:761]: How about "normalize" then?

Comment (tailhook) [reply edb/server/mng_port/edgecon.pyx]: Renamed.

**Pull request merged**